tdrs-frontend/src/components/Reports/Reports.jsx

```diff
@@ -27,61 +29,191 @@ function Reports() {
   const dispatch = useDispatch()
   const [isUploadReportToggled, setIsToggled] = useState(false)
 
-  const handleSelect = ({ target: { value } }) => {
-    setIsToggled(false)
-    dispatch(setYear(value))
+  const [formValidation, setFormValidationState] = useState({})
+  const [touched, setTouched] = useState({})
+
+  const quarters = {
+    Q1: 'Quarter 1 (October - December)',
+    Q2: 'Quarter 2 (January - March)',
+    Q3: 'Quarter 3 (April - June)',
+    Q4: 'Quarter 4 (July - September)',
   }
 
   const handleSearch = () => {
-    setIsToggled(true)
+    if (!isUploadReportToggled) {
+      // Clear previous errors
+      setFormValidationState({})
+      // Filter out non-truthy values
+      const form = [selectedYear, selectedStt, selectedQuarter].filter(Boolean)
+
+      if (form.length === 3) {
+        setIsToggled(true)
+      } else {
+        // create error state
+        setFormValidationState({
+          year: !selectedYear,
+          stt: !selectedStt,
+          quarter: !selectedQuarter,
+          errors: 3 - form.length,
+        })
+        setTouched({
+          year: true,
+          stt: true,
+          quarter: true,
+        })
+      }
+    }
+  }
+
```

Comment:
Handle the initial form errors and then "touch" the form to correct the validation in the `useEffect` below. The purpose of this is to prevent the form from "erroring" until it has been attempted to be filled out by the user.